doc/book/cookbook/setting-locale-depending-routing-parameter.md

```diff
@@ -108,22 +108,35 @@ registered in the pipeline immediately following the routing middleware.
 Such a `LocalizationMiddleware` class could look similar to this:
 
 ```php
+<?php
+
 namespace Application\I18n;
 
 use Locale;
-use Zend\Expressive\Router\RouteResult;
+use Psr\Http\Message\ResponseInterface;
+use Psr\Http\Message\ServerRequestInterface;
 
 class LocalizationMiddleware
 {
-    public function __invoke($request, $response, $next)
+    const LOCALIZATION_ATTRIBUTE = 'locale';
+
+    public function __invoke(ServerRequestInterface $request, ResponseInterface $response, callable $next = null)
     {
-        $locale = $request->getAttribute('locale', 'de_DE');
-        Locale::setDefault($locale);
-        return $next($request, $response);
+        // Get locale from route, fallback to the user's browser preference
+        $locale = $request->getAttribute(
+            'locale',
+            Locale::acceptFromHttp($request->getServerParams()['HTTP_ACCEPT_LANGUAGE'] ?? 'en_US')
```

Comment:
What happens if we write this as `current($request->getHeader('Accept-language'))` here? Also [Null Coalesce Operator](https://wiki.php.net/rfc/isset_ternary) (`??`) is available only PHP-7 and later, it worth to be noted in comments imho.